### PR TITLE
Mark Widget::destroy as unsafe

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #957.

cc @sdroege @EPashkin

### CODE REVIEW

File: src/widget.rs [hunk 1]
```diff
@@ -236,4 +238,10 @@ impl<O: IsA<Widget>> WidgetExtManual for O {
             gtk_sys::gtk_widget_set_events(self.as_ref().to_glib_none().0, events.to_glib() as i32);
         }
     }
+
+    unsafe fn destroy(&self) {
+        unsafe {
```
Comment (EPashkin) [inline src/widget.rs]:
```
warning: unnecessary `unsafe` block
   --> src\widget.rs:243:9
    |
243 |         unsafe {
    |         ^^^^^^ unnecessary `unsafe` block
    |
    = note: `#[warn(unused_unsafe)]` on by default
```

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks.
@sdroege How we implement closing application now?
Just changing to `unsafe{window.destroy();}`?
```
   Compiling gtk-rs-examples v0.0.1 (C:\projects\gtk\_examples)
error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> src\bin\menu_bar_system.rs:86:9
   |
86 |         window.destroy();
   |         ^^^^^^^^^^^^^^^^ call to unsafe function
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
error: aborting due to previous error
```

**Comment by sdroege** [comment]
>@sdroege How we implement closing application now?
>Just changing to `unsafe{window.destroy();}`?
>```

Just window.close() :)

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by sdroege** [comment]
Looks good to me apart from https://github.com/gtk-rs/gtk/pull/958#discussion_r381716230

And we should first update the examples so that everything continues to compile. Every `destroy()` call should be replaced with `close()`.

**Comment by GuillaumeGomez** [comment]
Updated and waiting for https://github.com/gtk-rs/examples/pull/296 now.

**Pull request merged**